test/tst_endian.py

```diff
@@ -65,6 +65,26 @@ def issue310(file):
                               var_big_endian[:].filled())
     nc.close()
 
+def issue346(file):
+    # create a big and a little endian variable
+    xb = np.arange(10, dtype='>i4')
+    xl = np.arange(xb.size, dtype='<i4')
+    nc = netCDF4.Dataset(file, mode='w')
+    nc.createDimension('x', size=xb.size)
+    vb=nc.createVariable('xb', xb.dtype, ('x'),
+                         endian='big')
+    vl=nc.createVariable('xl', xl.dtype, ('x'),
```

Comment:
What happens if a variable is created with endian='little' but big endian values are assigned to it?